tools/dependency_summary.py

```diff
@@ -0,0 +1,844 @@
+        "repository": "https://www.zlib.net/",
+        "license": "EXT-ZLIB",
+        "license_text": "This software makes use of the 'zlib' compression library, and we are very"\
+                        " grateful to Jean-loup Gailly, Mark Adler and team for producing it.",
+    },
+}
+
+# And these are rust packages that pull in the above dependencies.
+# Others are added on a per-target basis during dependency resolution.
+PACKAGES_WITH_EXTRA_DEPENDENCIES = {
+    "adler32": ["ext-zlib"],
+    "nss_sys": ["ext-nss", "ext-nspr"],
+    "openssl-sys": ["ext-openssl"],
+    "rusqlite": ["ext-sqlite"],
+    # Our `rc_crypto` crate copies the API of `ring`, so take a fake dependnecy
+    # in order to reflect accurate license information.
+    "rc_crypto": ["ext-ring"],
+    # As a special case, we know that the "logins" crate is the only thing that enables SQLCipher.
+    # In a future iteration we could check the cargo build-plan output to see whether anything is
+    # enabling the sqlcipher feature, but this will do for now.
+    "logins": ["ext-sqlcipher"],
+}
+
+# Hand-audited tweaks to package metadata, for cases where the data given to us by cargo is insufficient.
+# Let's try not to add any more dependencies that require us to edit this list!
+#
+# For each field we want to tweak, we list both the expected value from `cargo metadata` and the replacement
+# value we want to apply, like this:
+#
+#  {
+#    "example-package": {
+#      "license": {         # <-- the field we want to tweak
+#         "check": None     # <-- the value from `cargo metadata` (in this case, check that it's empty)
+#         "fixup": "MIT"    # <-- the value we want to replace it with
+#      }
+#    }
+#  }
+#
+# This is designed to prevent us from accidentally overriting future upstream changes in package metadata.
+# Any such changes will prevent us from regenerating the dependency summary, triggering a CI failure
+# and causing a human to investigate and resolve the change.
+PACKAGE_METADATA_FIXUPS = {
+    # In this case the rust code is BSD-3-Clause and the wrapped zlib library is under the Zlib license,
+    # which does not require explicit attribution.
+    "adler32": {
+        "license": {
+            "check": "BSD-3-Clause AND Zlib",
+            "fixup": "BSD-3-Clause",
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "LICENSE",
+        }
+    },
+    # These packages do not unambiguously delcare their licensing file.
+    "publicsuffix": {
+        "license": {
+            "check": "MIT/Apache-2.0"
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "LICENSE-APACHE",
+        }
+    },
+    "siphasher": {
+        "license": {
+            "check": "MIT/Apache-2.0"
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "COPYING",
+        }
+    },
+    # These packages do not include their license file in their release distributions,
```

Comment:
It's unfortunate we need this, but yeah, I don't see an alternative. Seems like it might add a lot of review churn, unfortunately.